components/script/dom/htmlvideoelement.rs

```diff
@@ -113,6 +114,11 @@ impl HTMLVideoElement {
         // Step 3.
         let poster_url = match ServoUrl::parse(poster_url) {
             Ok(url) => url,
+            Err(url::ParseError::RelativeUrlWithoutBase) => {
```

Comment:
step 3 of https://html.spec.whatwg.org/multipage/media.html#poster-frame says "parse relative to the node document", so we should match against `document_from_node(self).url().join(&poster_url)` instead; then we don't need to handle this case differently than any other error.